fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/AggregationAverageMaker.java

```diff
@@ -115,80 +125,72 @@ private TemplateDruidQuery buildOuterQuery(
      * @return A template query representing the inner aggregation
      */
     private TemplateDruidQuery buildInnerQuery(MetricField sourceMetric, TemplateDruidQuery innerDependentQuery) {
-
         Set<Aggregation> newInnerAggregations = convertSketchesToSketchMerges(innerDependentQuery.getAggregations());
-        Set<PostAggregation> newInnerPostAggregations = new LinkedHashSet<>();
 
-        // If the sourceMetric is a Post Aggregator, we need to preserve it in the inner query
-        if (sourceMetric instanceof PostAggregation) {
-            newInnerPostAggregations.add((PostAggregation) sourceMetric);
-        }
+        Set<PostAggregation> newInnerPostAggregations = (sourceMetric instanceof PostAggregation) ?
+                ImmutableSet.of((PostAggregation) sourceMetric) :
+                Collections.emptySet();
 
         // Build the inner query with the new aggregations and with the count
-        TemplateDruidQuery innerQuery = innerDependentQuery;
-        innerQuery = innerQuery.withAggregations(newInnerAggregations);
-        innerQuery = innerQuery.withPostAggregations(newInnerPostAggregations);
-        innerQuery = innerQuery.merge(buildTimeGrainCounterQuery());
-        return innerQuery;
+        return innerDependentQuery.withAggregations(newInnerAggregations)
+                .withPostAggregations(newInnerPostAggregations)
+                .merge(buildTimeGrainCounterQuery());
     }
 
     /**
-     * If the aggregation being averaged is a sketch, it will become a sketch merge, and need to be estimated for
-     * summing to work at the outer level.
+     * If the aggregation being averaged is a sketch, the inner query must convert it to a numerical type so that it
+     * can be summed.
      *
      * @param originalSourceMetric  The metric being target for sums
      *
      * @return Either the original MetricField, or a new SketchEstimate post aggregation
      */
     private MetricField convertToSketchEstimateIfNeeded(MetricField originalSourceMetric) {
-        if (originalSourceMetric instanceof SketchAggregation) {
-            return FieldConverterSupplier.sketchConverter.asSketchEstimate((SketchAggregation) originalSourceMetric);
-        }
-        return originalSourceMetric;
+        return originalSourceMetric instanceof SketchAggregation ?
+                sketchConverter.asSketchEstimate((SketchAggregation) originalSourceMetric) :
+                originalSourceMetric;
     }
 
     /**
      * Copy a set of aggregations, replacing any sketch aggregations with sketchMerge aggregations.
+     * This is an artifact of earlier sketch code which didn't have a single sketch type that could be used without
+     * finalization in inner queries.
```

Comment:
If that's the case, can we `@Deprecate` this method as well, so that when we do the next sweep to remove deprecated code, this will be flagged and we can remove it?